### PR TITLE
`YOLO` pipeline integration for deepsparse

### PR DESCRIPTION
**Pull request by rahul-tuli**
The goal of this PR is to add a pipeline interface for YOLO image segmentation models. The idea is to wrap preprocessing and postprocessing steps into a callable pipeline for ease of use, and out of the box support with `deepsparse.server`. This also adds an installable for yolo, to install all dependencies invoke: 
`pip install --editable "./[yolo]"`

Usage:  

- [X] Option 1, without a class names mapping

```python
from deepsparse.pipeline import Pipeline

model_stub = "zoo:cv/detection/yolov5-s/pytorch/ultralytics/coco/pruned-aggressive_96"
task = "yolo"
input_file = "yolo-test-image.JPEG"



pipeline = Pipeline.create(
    task=task,
    model_path=stub,
    class_names="coco",
)

print("Pipeline outputs: ", pipeline(images=input_file))
```

output:
```bash
DeepSparse Engine, Copyright 2021-present / Neuralmagic, Inc. version: 0.11.2 (ca6e29a5) (release) (optimized) (system=avx512, binary=avx512)
Pipeline outputs:  predictions=[[[132.24066162109375, 0.04663848876953125, 269.58807373046875, 46.56146240234375, 0.4553620517253876, 2.0], [181.0561065673828, 7.16241455078125, 447.66375732421875, 612.1524047851562, 0.444583535194397, 0.0]]] boxes=[[[132.24066162109375, 0.04663848876953125, 269.58807373046875, 46.56146240234375], [181.0561065673828, 7.16241455078125, 447.66375732421875, 612.1524047851562]]] scores=[[0.4553620517253876, 0.444583535194397]] labels=[['car', 'person']]

```

### CODE REVIEW

File: src/deepsparse/yolo/utils.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+
+try:
+    import cv2
+
+    cv2_error = None
+except ModuleNotFoundError as cv2_import_error:
+    cv2 = None
+    cv2_error = cv2_import_error
+
+COCO_CLASSES = [
```
Comment (dbogunowicz) [inline src/deepsparse/yolo/utils.py:38]: Maybe keeping this list in a separate file would be more elegant?

**Review by dbogunowicz — COMMENTED** [review]
lgtm, good job. I would refactor some of the code, but I assume that those pieces which I do not like are copied from the original yolo codebase, so it does not make sense to beautify it - better to have it in the original form for reference, would you agree?

File: src/deepsparse/yolo/utils.py (hunk 1 above)
Comment (rahul-tuli) [reply src/deepsparse/yolo/utils.py:38]: > lgtm, good job. I would refactor some of the code, but I assume that those pieces which I do not like are copied from the original yolo codebase, so it does not make sense to beautify it - better to have it in the original form for reference, would you agree?

I agree a future PR would refactor that "copied" code out to use our own hosted YOLO wheel

**Review by dbogunowicz — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Comment by bfineran** [comment]
rebased onto feature-pipeline-main

**Pull request merged**